File: WdtcCore/src/main/java/org/wdt/wdtc/core/launch/GameClassPath.java
package org.wdt.wdtc.core.launch;


import org.apache.log4j.Logger;
import org.wdt.utils.io.FileUtils;
import org.wdt.wdtc.core.game.GetGameNeedLibraryFile;
import org.wdt.wdtc.core.game.Launcher;
import org.wdt.wdtc.core.manger.FileManger;
import org.wdt.wdtc.core.manger.SettingManger;
import org.wdt.wdtc.core.utils.WdtcLogger;
import org.wdt.wdtc.core.utils.ZipUtils;

import java.io.IOException;

public class GameClassPath extends AbstractGameCommand {
  private static final Logger logmaker = WdtcLogger.getLogger(GameClassPath.class);
  private final Launcher launcher;

  public GameClassPath(Launcher launcher) {
    this.launcher = launcher;
  }

  public StringBuilder getCommand() {
    try {
      GameLibraryData gameLibraryData = new GameLibraryData(launcher);
      FileUtils.createDirectories(launcher.getVersionNativesPath());
      GetGameNeedLibraryFile FileList = new GetGameNeedLibraryFile(launcher);
      for (GetGameNeedLibraryFile.LibraryFile LibraryFile : FileList.getFileList()) {
        if (LibraryFile.isNativesLibrary()) {
          ZipUtils.unzipByFile(gameLibraryData.GetNativesLibraryFile(LibraryFile.getLibraryObject().getDownloads().getClassifiers().getNativesindows()), FileUtils.getCanonicalPath(launcher.getVersionNativesPath()));
        } else {
          insertclasspathSeparator(gameLibraryData.GetLibraryFile(LibraryFile.getLibraryObject()));
        }
      }
      command.append(launcher.getVersionJar());
      String Accounts = launcher.getAccounts().getJvmCommand();
      if (!Accounts.isEmpty()) {
        command.append(Accounts);
      }
      if (SettingManger.getSetting().isLlvmpipeLoader()) {
        command.append(LlbmpipeLoader());
      }

    } catch (IOException e) {
      logmaker.error(WdtcLogger.getExceptionMessage(e));
    }
    return command;
  }


  private String LlbmpipeLoader() {
    return " -javaagent:" + FileManger.getLlbmpipeLoader();
  }
}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/launch/GameClassPath.kt
package org.wdt.wdtc.core.launch

import org.wdt.utils.io.createDirectories
import org.wdt.wdtc.core.auth.accounts.Accounts
import org.wdt.wdtc.core.game.GetGameNeedLibraryFile
import org.wdt.wdtc.core.game.Launcher
import org.wdt.wdtc.core.manger.llbmpipeLoader
import org.wdt.wdtc.core.manger.setting
import org.wdt.wdtc.core.utils.getExceptionMessage
import org.wdt.wdtc.core.utils.logmaker
import org.wdt.wdtc.core.utils.unzipByFile
import java.io.IOException

class GameClassPath(private val launcher: Launcher) : AbstractGameCommand() {
  override fun getCommand(): StringBuilder {
    try {
      val gameLibraryData = GameLibraryData(launcher)
      launcher.versionNativesPath.createDirectories()
      val fileList = GetGameNeedLibraryFile(launcher)
      for (libraryFile in fileList.libraryList) {
        if (libraryFile.nativesLibrary) {
          unzipByFile(
            gameLibraryData.getNativesLibraryFile(libraryFile.libraryObject.downloads?.classifiers?.nativesindows!!),
            launcher.versionNativesPath.canonicalPath
          )
        } else {
          insertclasspathSeparator(gameLibraryData.getLibraryFile(libraryFile.libraryObject))
        }
      }
      commandBuilder.append(launcher.versionJar)
      val accounts = Accounts().jvmCommand
      if (accounts.isNotEmpty()) {
        commandBuilder.append(accounts)
      }
      if (setting.llvmpipeLoader) {
        commandBuilder.append(llbmpipeLoaderCommand)
      }
    } catch (e: IOException) {
      logmaker.error(e.getExceptionMessage())
    }
    return commandBuilder
  }

  private val llbmpipeLoaderCommand: String
    get() = " -javaagent:$llbmpipeLoader"

}
